src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -568,6 +688,14 @@ void GcsActorManager::OnWorkerDead(const ray::ClientID &node_id,
     }
   }
 
+  // The creator worker of these actors died before resolving their dependencies. In this
+  // case, these actors will never be created successfully. So we need to mark them dead,
+  // to prevent actor tasks hang forever.
+  auto unresolved_actors = CancelUnresolvedActors(node_id, worker_id);
+  for (auto &actor_id : unresolved_actors) {
+    ReconstructActor(actor_id, /*need_reschedule=*/false);
```

Comment:
Shouldn't we call `DestroyActor`? I don't think ReconstructActor cleans up the local state.

Reply:
> Shouldn't we call `DestroyActor`? I don't think ReconstructActor cleans up the local state.

Yes, my bad, I will fix it.